imagebuildah/stage_executor.go

```diff
@@ -595,6 +596,24 @@ func (s *StageExecutor) prepare(ctx context.Context, from string, initializeIBCo
 		}
 	}
 
+	builderSystemContext := s.executor.systemContext
+	// get platform string from stage
+	if stage.Builder.Platform != "" {
+		os, arch, variant, err := parse.Platform(stage.Builder.Platform)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to parse platform %q", stage.Builder.Platform)
+		}
+		if arch != "" {
+			builderSystemContext.ArchitectureChoice = arch
+		}
+		if os != "" {
+			builderSystemContext.OSChoice = os
+		}
+		if variant != "" {
+			builderSystemContext.VariantChoice = variant
+		}
```

Comment:
`arch` and `variant` need to be managed together, else on a `linux/arm/v7` system or the like you could conceivably find yourself attempting to pull an image for `linux/386/v7`.

Reply:
fixed.